tests/unit/basic_crawler/test_basic_crawler.py

```diff
@@ -20,15 +20,34 @@
 from crawlee.basic_crawler import BasicCrawler
 from crawlee.configuration import Configuration
 from crawlee.errors import SessionError, UserDefinedErrorHandlerError
+from crawlee.events import EventManager
 from crawlee.statistics import FinalStatistics
 from crawlee.storages import Dataset, KeyValueStore, RequestList, RequestQueue
 
 if TYPE_CHECKING:
-    from collections.abc import Sequence
+    from collections.abc import AsyncGenerator, Sequence
 
     import respx
     from yarl import URL
 
+    from crawlee._types import JsonSerializable
+
+
+@pytest.fixture
+async def mock_event_manager() -> AsyncGenerator[EventManager, None]:
+    async with EventManager(persist_state_interval=timedelta(milliseconds=50)) as event_manager:
+        with patch('crawlee.service_container.get_event_manager', return_value=event_manager):
+            yield event_manager
+
+
+@pytest.fixture
+async def key_value_store() -> AsyncGenerator[KeyValueStore, None]:
+    kvs = await KeyValueStore.open()
+    yield kvs
+    await kvs.drop()
+    kvs._clear_cache()
+    kvs._drop_persist_state_event()
```

Comment:
Might be better to use a mock event manager and just discard it at the end of the test. That way the `KeyValueStore` doesn't have to deal with unregistering the listeners at all, and I think that's a good thing.